apps/ledger-live-mobile/src/hooks/useAnalyticsOptInPromptVariantB.ts

```diff
@@ -23,10 +23,13 @@ const useAnalyticsOptInPrompt = () => {
     useNavigation<
       RootNavigationComposite<StackNavigatorNavigation<OnboardingNavigatorParamList>>
     >();
-  const analyticsEnabled: boolean = useSelector(analyticsEnabledSelector);
-  const personalizedRecommendationsEnabled: boolean = useSelector(
+  const analyticsEnabled = useSelector(analyticsEnabledSelector);
+  const personalizedRecommendationsEnabled = useSelector(
     personalizedRecommendationsEnabledSelector,
   );
+  const isTrackingEnabled = analyticsEnabled || personalizedRecommendationsEnabled;
+  // When the user has not made a choice yet, we can track the analytics opt in flow
+  const shouldWeTrack = isTrackingEnabled === true || isTrackingEnabled === null;
```

Comment:
```suggestion
const shouldWeTrack = !!isTrackingEnabled 
```

Reply:
We need it to return true whether the isTrackingEnabled value is true or null, casting the value to a boolean won't work here